webrender/examples/multiwindow.rs

```diff
@@ -0,0 +1,282 @@
+        let api = sender.create_api();
+        let document_id = api.add_document(framebuffer_size, 0);
+
+        let epoch = Epoch(0);
+        let pipeline_id = PipelineId(0, 0);
+        let mut resources = ResourceUpdates::new();
+
+        let font_key = api.generate_font_key();
+        let font_bytes = load_file("../wrench/reftests/text/FreeSans.ttf");
+        resources.add_raw_font(font_key, font_bytes, 0);
+
+        let font_instance_key = api.generate_font_instance_key();
+        resources.add_font_instance(font_instance_key, font_key, Au::from_px(32), None, None, Vec::new());
+
+        let mut txn = Transaction::new();
+        txn.update_resources(resources);
+        api.send_transaction(document_id, txn);
+
+        Window {
+            window,
+            renderer,
+            name,
+            epoch,
+            pipeline_id,
+            document_id,
+            api,
+            font_instance_key,
+        }
+    }
+
+    fn tick(&mut self) -> bool {
+        unsafe {
+            self.window.make_current().ok();
+        }
+
+        for event in self.window.poll_events() {
+            match event {
+                glutin::Event::Closed |
+                glutin::Event::KeyboardInput(_, _, Some(glutin::VirtualKeyCode::Escape)) => return true,
+
+                glutin::Event::KeyboardInput(
+                    glutin::ElementState::Pressed,
+                    _,
+                    Some(glutin::VirtualKeyCode::P),
+                ) => {
+                    println!("toggle flags {}", self.name);
+                    self.renderer.toggle_debug_flags(webrender::DebugFlags::PROFILER_DBG);
+                }
+
+                _ => {}
+            }
+        }
+
+        let framebuffer_size = {
+            let (width, height) = self.window.get_inner_size_pixels().unwrap();
+            DeviceUintSize::new(width, height)
+        };
+        let device_pixel_ratio = self.window.hidpi_factor();
+        let layout_size = framebuffer_size.to_f32() / euclid::TypedScale::new(device_pixel_ratio);
+        let mut txn = Transaction::new();
+        let mut builder = DisplayListBuilder::new(self.pipeline_id, layout_size);
+
+        let bounds = LayoutRect::new(LayoutPoint::zero(), builder.content_size());
+        let info = LayoutPrimitiveInfo::new(bounds);
+        builder.push_stacking_context(
+            &info,
+            ScrollPolicy::Scrollable,
+            None,
+            TransformStyle::Flat,
+            None,
+            MixBlendMode::Normal,
+            Vec::new(),
+        );
+
+        let info = LayoutPrimitiveInfo::new(LayoutRect::new(LayoutPoint::new(100.0, 100.0), LayoutSize::new(100.0, 200.0)));
+        builder.push_rect(&info, ColorF::new(0.0, 1.0, 0.0, 1.0));
+
+        let text_bounds = LayoutRect::new(LayoutPoint::new(100.0, 50.0), LayoutSize::new(700.0, 200.0));
+        let glyphs = vec![
```

Comment:
let's remove the text from our basic example to avoid duplication?

Reply:
Done